exporter/coralogixexporter/traces_client.go

```diff
@@ -64,9 +64,22 @@ func (e *tracesExporter) pushTraces(ctx context.Context, td ptrace.Traces) error
 
 	partialSuccess := resp.PartialSuccess()
 	if partialSuccess.ErrorMessage() != "" || partialSuccess.RejectedSpans() != 0 {
+		var traceIDs []string
+		rss := td.ResourceSpans()
+		for i := 0; i < rss.Len(); i++ {
+			ss := rss.At(i).ScopeSpans()
+			for j := 0; j < ss.Len(); j++ {
+				spans := ss.At(j).Spans()
+				for k := 0; k < spans.Len(); k++ {
+					traceIDs = append(traceIDs, spans.At(k).TraceID().String())
+				}
+			}
```

Comment:
Isn't looping over all spans slightly expensive? Maybe we could accumulate the trace IDs somewhere as the spans arrive at the exporter instead? 🤔

Reply:
We talked offline and we agreed to do this only when the debug level is set in the logger.

Also, I noticed about some small improvements that can be done.